// tweeter-shared/src/index.ts
//
// Domain Classes
//
export { Follow } from "./model/domain/Follow";
export { PostSegment, Type } from "./model/domain/PostSegment";
export { Status } from "./model/domain/Status";
export { User } from "./model/domain/User";
export { AuthToken } from "./model/domain/AuthToken";

// All classes that should be avaialble to other modules need to exported here. export * does not work when 
// uploading to lambda. Instead we have to list each export.
//
// Other
//
export { FakeData } from "./util/FakeData";

//
// DTOs
//
export type { UserDto } from "./model/dto/UserDto";
export type { StatusDto } from "./model/dto/StatusDto";

//
// Requests
//
export type { PagedUserItemRequest } from "./model/domain/net/request/PagedUserItemRequest";
export type { FollowCountRequest } from "./model/domain/net/request/FollowCountRequest";
export type { FollowRequest } from "./model/domain/net/request/FollowRequest";
export type { GetIsFollowerStatusRequest } from "./model/domain/net/request/GetIsFollowerStatusRequest";
export type { LoadMoreItemsRequest } from "./model/domain/net/request/LoadMoreItemsRequest";
export type { PostStatusRequest } from "./model/domain/net/request/PostStatusRequest";
export type { GetUserRequest } from "./model/domain/net/request/GetUserRequest";
export type { LoginRequest } from "./model/domain/net/request/LoginRequest";
export type { RegisterRequest }from "./model/domain/net/request/RegisterRequest";
export type { LogoutRequest }from "./model/domain/net/request/LogoutRequest";
export type { TweeterRequest } from "./model/domain/net/request/TweeterRequest";
//
// Responses
//
export type { TweeterResponse } from "./model/domain/net/response/TweeterResponse";
export type { PagedUserItemResponse } from "./model/domain/net/response/PagedUserItemResponse";
export type { FollowCountResponse } from "./model/domain/net/response/FollowCountResponse";
export type { FollowResponse } from "./model/domain/net/response/FollowResponse";
export type { GetIsFollowerStatusResponse } from "./model/domain/net/response/GetIsFollowerStatusResponse";
export type { LoadMoreItemsResponse } from "./model/domain/net/response/LoadMoreItemsResponse";
export type { PostStatusResponse } from "./model/domain/net/response/PostStatusResponse";
export type { GetUserResponse } from "./model/domain/net/response/GetUserResponse";
export type { LoginResponse } from "./model/domain/net/response/LoginResponse";
export type { LogoutResponse }from "./model/domain/net/response/LogoutResponse";
